docs/running-clusterfuzzlite/gitlab.md

```diff
@@ -0,0 +1,229 @@
+
+Docker-in-Docker does not seem possible as clusterfuzz-lite images
+do not have the docker tools installed.
+
+## .gitlab-ci.yml
+For basic ClusterFuzzLite functionality, all you need is a single job
+to enable fuzzing on your merge requests.
+
+To enable more features, we recommend having different jobs for:
+
+- continuous builds
+- batch fuzzing and corpus pruning
+- coverage
+
+### MR fuzzing
+
+To add a fuzzing job that fuzzes all merge requests to your repo, add the
+following default configurations to `.gitlab-ci.yml`:
+
+{% raw %}
+```yaml
+variables:
+  SANITIZER: "address"
+  CFL_PLATFORM: gitlab
+
+clusterfuzzlite:
+  image:
+    name: google/clusterfuzzlite/actions/build_fuzzers@v1
+    entrypoint: [""]
+  stage: fuzz
+  rules:
+    # default code change
+    - if: $CI_PIPELINE_SOURCE == "merge_request_event"
+      variables:
+        MODE: "code-change"
+  before_script:
+    # gitlab's container name is not its hostname !
+    - export CFL_CONTAINER_ID=`cut -c9- < /proc/1/cpuset`
+  script:
+    # will build and run the fuzzers
+    - python3 "/opt/oss-fuzz/infra/cifuzz/cifuzz_combined_entrypoint.py"
+  artifacts:
+    # upload artifacts when a crash makes the job fail
+    when: always
+    paths:
+      - artifacts/crashes/
+```
+{% endraw %}
+
+You may also wish to set [tags](https://docs.gitlab.com/runner/#tags) to select a relevant runner.
+
+Optionally, edit the following variables to customize your settings:
+- `SANITIZER` Change or enable more sanitizers.
+- `LANGUAGE` Define the language of your project.
+- `CFL_BRANCH` Branch to fuzz, default is `CI_DEFAULT_BRANCH`.
+- `FILESTORE` To use corpus produced by other jobs.
+- `FUZZ_SECONDS` Change the amount of time spent fuzzing.
+- `CFL_ARTIFACTS_DIR` : To save your artifacts in a different directory than `artifacts`
+
+For `SANITIZER`, you may also use a matrix to use multiple sanitizers with the same job.
+
+### Batch fuzzing and corpus pruning
+
+Batch fuzzing enables continuous, regular fuzzing on your latest HEAD and
+allows a corpus of inputs to build up over time, which greatly improves the
+effectiveness of fuzzing. Batch fuzzing should be run on a schedule.
+
+To enable batch fuzzing, add the following to
+`.gitlab-ci.yml`:
+
+{% raw %}
+```yaml
+# this name is hardcoded and cannot be changed for gitlab artifacts filestore
+clusterfuzzlite-corpus:
+  image:
+    name: google/clusterfuzzlite/actions/build_fuzzers@v1
+    entrypoint: [""]
+  stage: fuzz
+  rules:
+    - if: $MODE == "prune"
+    - if: $MODE == "batch"
+  before_script:
+    # gitlab's container name is not its hostname !
+    - export CFL_CONTAINER_ID=`cut -c9- < /proc/1/cpuset`
+  script:
+    - python3 "/opt/oss-fuzz/infra/cifuzz/cifuzz_combined_entrypoint.py"
+  artifacts:
+    when: always
+    paths:
+      - artifacts/corpus/
+      - artifacts/crashes/
+```
+{% endraw %}
+
+You should then define two [schedules](https://docs.gitlab.com/ee/ci/pipelines/schedules.html)
```

Comment:
Maybe have some screenshots walk through an example of setting this up.

Reply:
I can do that once the rest is good for you

Reply:
Sure!